erizo_controller/erizoJS/erizoJSController.js

```diff
@@ -100,18 +100,18 @@ exports.ErizoJSController = (threadPool, ioThreadPool) => {
 
     const closePromise = node.close(sendOffer);
 
-    const client = clients.get(clientId);
-    if (client === undefined) {
-      log.debug('message: trying to close node with no associated client,' +
-        `clientId: ${clientId}, streamId: ${node.streamId}`);
-      return Promise.resolve();
-    }
-
-    const remainingConnections = client.maybeCloseConnection(connection.id);
-    if (remainingConnections === 0) {
-      log.debug(`message: Client is empty, clientId: ${client.id}`);
-    }
-    return closePromise;
+    return closePromise.then(() => {
+      const client = clients.get(clientId);
+      if (client === undefined) {
+        log.debug('message: trying to close node with no associated client,' +
+          `clientId: ${clientId}, streamId: ${node.streamId}`);
+        return;
+      }
+      const remainingConnections = client.maybeCloseConnection(connection.id);
```

Comment:
heads up: this could result in new issues given that maybeCloseConnection is now async

Reply:
I agree, and we will have to test and monitor this. However, having the WebRTCConnection object being destroyed because of maybeCloseConnection while removeStream is potentially still running is also not ideal.